### PR TITLE
feat(cli): graceful close of vitest in watch mode

### PR DESCRIPTION
**Pull request by dominikg**
This change ensures that vite buildEnd hook is called and resources are cleaned up gracefully

### CODE REVIEW

**Review by patak-dev — APPROVED** [review]

File: packages/vitest/src/node/index.ts [hunk 1]
```diff
@@ -248,8 +250,17 @@ class Vitest {
   }
 
   async close() {
```
Comment (dammy001) [inline packages/vitest/src/node/index.ts:250]: async without await here

Comment (dominikg) [reply packages/vitest/src/node/index.ts:250]: it is not required that async functions contain an await and close returns a promise, so the contract of async is there.
And while it is not strictly needed here i think it doesn't hurt and shows the async nature of the implementation.

**Comment by userquin** [comment]
`nr ci` + `pnpm run test:ci` + `pnpm run test:all` working on windows, no memory error reported (6 consecutive runs)

**Comment by userquin** [comment]
another 6 consecutives runs with latest changes with no memory error reported on windows

**Pull request merged**